src/api_client.rs

```diff
@@ -292,8 +287,22 @@ impl Client {
         unimplemented!("")
     }
 
-    pub async fn get_node_version() -> Result<String, Error> {
-        unimplemented!("")
+    pub async fn get_node_version(&self) -> String{
```

Comment:
so many things could fail in this function and we want to give the caller a chance to respond, rather than just incur a panic, like would currently happen w/ all the `unwrap`s

```suggestion
    pub async fn get_node_version(&self) -> Result<String, Error> {
```